### PR TITLE
Fix up the docs

### PR DESCRIPTION
**Pull request by tannewt**
* Fix badges
* Remove const from address so it doesn't show as mock object.
* Use rst style params
* Mock micropython import
* Remove "Read" from properties

### CODE REVIEW

**Comment by tannewt** [comment]
I fixed up the lint nitpicks. Please take a look.

File: adafruit_tsl2591.py [hunk 1]
```diff
@@ -39,7 +39,7 @@
 
 #pylint: disable=bad-whitespace
 # Internal constants:
-_TSL2591_ADDR                = const(0x29)
+_TSL2591_ADDR                = 0x29
```
Comment (tannewt) [inline adafruit_tsl2591.py:42]: From an email thread with @tdicola:

```
why the change from a const for the I2C address to non-const value in all these pulls? 
That's a small (and admittedly low risk) change to the function of the code, not just the docs.
```

Good point! This improves how the default value is rendered by sphinx. With the `const()` it ends up as a MockObject to sphinx like [this](https://circuitpython.readthedocs.io/projects/bme280/en/latest/api.html#adafruit_bme280.Adafruit_BME280_I2C). Changing it to a regular number means it shows the decimal value. We could look into mocking `const `out properly instead.

**Comment by tannewt** [comment]
Fixed some of my comments to not mention returning anything. Also added a `light` property to match the sensor guide doc here: https://circuitpython.readthedocs.io/en/2.x/docs/design_guide.html#sensor-properties-and-units

**Comment by tdicola** [comment]
Please send functional changes like new properties as a new pull.  Those definitely need explicit validation on the hardware.  It's good to keep a clear difference between pulls that modify docs vs. code so we know the risk of merging them and work required to validate.

File: adafruit_tsl2591.py (hunk 1 above)
Comment (tdicola) [reply adafruit_tsl2591.py:42]: Yeah the const choice is an explicit decision to reduce memory usage though.  I would rather we err on the side of efficiency.  Does sphinx have a way to manually override the generated doc in this case?  That would be the best option if we don't like the mock object there.

**Comment by ladyada** [comment]
waaaaait! :)

there's two properties, lux which is really lux, and light which is just monotonic light (a.k.a photocell) but not lux. TSL2591 (and other good sensors) give real Lux readings.

@tannewt its been a while but i remember we discussed this and updated it in master
https://github.com/adafruit/circuitpython/blob/master/docs/design_guide.rst#sensor-properties-and-units
so its here
http://circuitpython.readthedocs.io/en/latest/docs/design_guide.html (not 2.x)

if there are no compelling reasons otherwise, lets stick to the master design guide! (that is, lux)

**Comment by tdicola** [comment]
Ah great point, and yeah right now the lib is clear the lux property gives lux and the other luminosities are unitless.  In the guide it calls out for example: https://learn.adafruit.com/adafruit-tsl2591?view=all#circuitpython  "The visible light level measured by the sensor.  This is a 32-bit unsigned value with no units, the higher the number the more visible light."

**Comment by ladyada** [comment]
we *could* require all 'lux' sensors to also expose 'light' - that is as a superset property

if that's what y'all want, i'll update the design to indicate that

**Comment by tdicola** [comment]
Yeah I'm going to close this PR since there's too much going on with it.  I just separately pulled out changes and applied them to master so sphinx docs build without warning:
```
tony-imac:TSL2591 tony$ sphinx-build -E -W -b html . _build/html
Running Sphinx v1.6.3
loading intersphinx inventory from https://docs.python.org/3.4/objects.inv...
loading intersphinx inventory from https://circuitpython.readthedocs.io/projects/bus_device/en/latest/objects.inv...
loading intersphinx inventory from https://circuitpython.readthedocs.io/en/latest/objects.inv...
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 2 source files that are out of date
updating environment: 2 added, 0 changed, 0 removed
reading sources... [100%] api                                                                        
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] api                                                                         
generating indices... genindex py-modindex
highlighting module code... [100%] adafruit_tsl2591                                                  
writing additional pages... search
copying static files... done
copying extra files... done
dumping search index in English (code: en) ... done
dumping object inventory... done
build succeeded.
```
Verified on the hardware and lint is all good.  Any other changes can be sent as separate, isolated pulls.

File: adafruit_tsl2591.py (hunk 1 above)
Comment (tannewt) [reply adafruit_tsl2591.py:42]: Now that I think about it, a better option is probably having a pip dependency that simply has const as a lambda. I feel like this exists somewhere.

**Comment by tannewt** [comment]
@ladyada thanks for the note about the new properties! I need to be better about using latest.

@tdicola I'll follow up with other PRs as needed. Thanks for the sphinx fix!

**Pull request closed without merging**